src/components/dataSet-settings.js

```diff
@@ -83,22 +83,42 @@ class DataSetSettings extends React.Component {
         },
     }
 
+    chooseSetting = (name, value) => {
+        switch (name) {
+            case 'periodDSDownload':
+                value = parseInt(value)
```

Comment:
Rather than assigning to the `value` argument, why not just `return parseInt(value)` here?  That way you also don't need the `break` line